docs/cookbook/dynamic-configuration/01_Basics/03_Integrations.md

```diff
@@ -0,0 +1,43 @@
+# 03 Integrations
+
+Dynamic configuration requires the use of `integrations-next` feature flag, to allow arrays of integrations. In this we will load integrations of various types. This is all built on the previous examples.
+
+`docker run -v ${PWD}/:/etc/grafana grafana/agentctl:latest template-parse /etc/grafana/03_config.yml`
+
+## Dynamic Configuration
+
+[config.yml](./03_config.yml)
+
+Tells the Grafana Agent where to load files from.
+
+## Integrations
+
+Integrations are loaded from files matching `integrations-*.yml` and are combined together. You can declare for example multiple sets of `redis_exporter_configs` across several files. 
+
+[integrations-node.yml](./03_assets/integrations-node.yml)
+
+Note: You do NOT have to name the above file `integrations-node.yml` with `node`, `integrations-1.yml` would work the same. The name does NOT determine the type of integrations a template can contain and a template can contain integrations of different types.  
+
+```yaml
+node_exporter: {}
+```
+
+[integrations-redis.yml](./03_assets/integrations-redis.yml)
+
+```yaml
+redis_exporter_configs:
+  - redis_addr: localhost:6379
+    autoscrape:
+      metric_relabel_configs:
+        - source_labels: [__address__]
+          target_label: "banana"
+          replacement: "apple"
+  - redis_addr: localhost:6380
```

Comment:
I think this might be a pretty big problem for the "config snippet" concept. I'd argue it's a requirement that you're able to have one file per instantiation of a redis_exporter.

Reply:
You can but you need to include the `redis_exporter_configs` at the top of each file. The above could be split into two separate files with

```
redis_exporter_configs:
  - redis_addr: localhost:6379
    autoscrape:
      metric_relabel_configs:
        - source_labels: [__address__]
          target_label: "banana"
          replacement: "apple"
```

```
redis_exporter_configs: 
- redis_addr: localhost:6380
```

 The templates are additive except singletons, which throws an error. 

Does that address your concern?

Reply:
Yes, that'd be fine 👍